src/Native/Unix/System.Native/pal_random.c

```diff
@@ -40,7 +40,12 @@ void SystemNative_GetNonCryptographicallySecureRandomBytes(uint8_t* buffer, int3
 
             do
             {
+#ifdef O_CLOEXEC
```

Comment:
I would prefer detecting this in the configure.cmake too. The O_CLOEXEC doesn't have to be a define and in such case, this check would break.

Reply:
The nice thing about doing this in configure.cmake is that we can easily see in the build output / CI logs what choice it's making.  If this ifdef were to fail for some unexpected reason, the code would fall back to the worse path, and we wouldn't know it.